BundleTransport/app/src/main/java/net/discdd/bundletransport/UsbFragment.java

```diff
@@ -0,0 +1,229 @@
+    private TransportPaths transportPaths;
+    private Button usbExchangeButton;
+    private TextView usbConnectionText;
+    private UsbManager usbManager;
+    private BroadcastReceiver mUsbReceiver;
+    public static boolean usbConnected = false;
+    private StorageManager storageManager;
+    private static final Logger logger = Logger.getLogger(UsbFragment.class.getName());
+    private static final String usbDirName = "/DDD_transport";
+    private ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+    private File usbDirectory;
+    public Path transportPath;
+
+    public UsbFragment(TransportPaths transportPaths) {
+        this.transportPaths = transportPaths;
+    }
+
+    @Override
+    public View onCreateView(
+            @NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.usb_fragment, container, false);
+
+        usbExchangeButton = view.findViewById(R.id.usb_exchange_button);
+        usbConnectionText = view.findViewById(R.id.usbconnection_response_text);
+
+        usbManager = (UsbManager) getActivity().getSystemService(Context.USB_SERVICE);
+        mUsbReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                handleUsbBroadcast(intent);
+            }
+        };
+        storageManager = (StorageManager) getActivity().getSystemService(Context.STORAGE_SERVICE);
+
+        //Register USB broadcast receiver
+        IntentFilter filter = new IntentFilter();
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_DETACHED);
+        filter.addAction(UsbManager.ACTION_USB_DEVICE_ATTACHED);
+        getActivity().registerReceiver(mUsbReceiver, filter);
+
+        // Check initial USB connection
+        checkUsbConnection(3);
+
+        usbExchangeButton.setOnClickListener(v -> {
+            try {
+                populateUsb();
+            } catch (IOException e) {
+                logger.log(INFO, "Populate USB was unsuccessful");
+                throw new RuntimeException(e);
+            }
+            logger.log(INFO, "Sync button was hit");
+        });
+        return view;
+    }
+
+    /**
+     * Finds appropriate usb dir for ddd activities and copies device files.
+     *
+     * @throws IOException
+     */
+    private void populateUsb() throws IOException {
+        List<StorageVolume> storageVolumes = storageManager.getStorageVolumes();
+        for (StorageVolume volume : storageVolumes) {
+            if (volume.isRemovable() && !volume.isEmulated()) {
+                File usbStorageDir = volume.getDirectory();
+                if (usbStorageDir != null) {
+                    File dddTransportDir = new File(usbStorageDir, usbDirName);
+                    if (!dddTransportDir.exists()) {
+                        dddTransportDir = new File(usbStorageDir, "/DDD_transport/server");
```

Comment:
remove the leading / . you should also make these final static Strings at the top.

i don't think you use these paths when you actually do the copy